impl/src/planning/crate_catalog.rs

```diff
@@ -121,6 +121,8 @@ impl CrateCatalogEntry {
         "vendor/{}/{}",
         &self.package_ident, settings.output_buildfile_suffix,
       ),
+      // Settings should always have `genmode` set to one of the above fields
+      GenMode::Unspecified => unreachable!(),
```

Comment:
I'd actually prefer an explicit panic here (or local_build_path to be updated to return Result<>)

My second preference would be to explain in the comment why this is unreachable rather than just assert that it's unreachable (in fact, I don't think it is -- going through validate_settings may not be only conceivable path to this point).